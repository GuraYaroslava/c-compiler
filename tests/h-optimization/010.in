int main()
{
    int a = 24 / ((3 * ((5 + (36 + 4)) / 5)) % 5) >> 2;
    while (a)
    {
        a -= (24 / ((3 * ((5 + (36 + 4)) / 5)) % 5) >> 2) - 2;
        printf("bodyFor\n");
    }
    return 0;
}
