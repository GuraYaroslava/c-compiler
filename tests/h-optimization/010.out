table 0:
`int`
`float`
`char`

table 1:
variable `main` type of function() returning `int`
{
variable `a` type of `int`
        a
    = <
        3

while
    a
{
        a
    -= <
        1

    function:     printf
        format:     "bodyFor\n"


}
return
    0
}


